src/services/utilServices/MailClient.ts

```diff
@@ -28,16 +49,76 @@ class MailClient {
     }
 
     try {
-      await axios.post(endpoint, email, {
+      const sendMailResponse = await axios.post<MailData>(sendEndpoint, email, {
         headers: {
           Authorization: `Bearer ${this.POSTMAN_API_KEY}`,
         },
       })
+      this.verifyMail(sendMailResponse.data)
     } catch (err) {
       logger.error(`Error occurred when sending email to ${recipient}: ${err}`)
       throw new Error("Failed to send email.")
     }
   }
+
+  async verifyMail(sendMailData: MailData): Promise<void> {
+    const verifyEndpoint = `${POSTMAN_API_URL}/transactional/email/${sendMailData.id}`
+    await new Promise((res) => setTimeout(res, MAIL_VERIFICATION_DELAY))
+    const verifyMailResponse = await axios.get<MailData>(verifyEndpoint, {
+      headers: {
+        Authorization: `Bearer ${this.POSTMAN_API_KEY}`,
+      },
+    })
+    // NOTE: For tests where the main routine has already ended, mockAxios is terminated
+    // and will return undefined for verifyMailResponse. In order to pass the test, we
+    // need to check if it is a valid response. This should not happen in application.
+    if (verifyMailResponse) {
+      this.logEmailStatus(verifyMailResponse.data)
+    }
+  }
+
+  logEmailStatus(verifyMailData: MailData) {
+    const mailStatus = verifyMailData.status
+    let unknownMailStatus: never
```

Comment:
nit: assign in line in the default case for locality of information

Reply:
Will have eslint warning for declaration in switch case

Reply:
will result in eslint warning for declaration of variable in switch case